### PR TITLE
Add InitialDataTci to ValenciaDivClean

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

exactly as the description says. I don't expect the ID TCI to change much or at all compared to the evolution TCIs where we'll need to gain some experience with different shock interactions, etc.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/InitialDataTci.hpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "DataStructures/VariablesTag.hpp"
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/InitialDataTci.hpp]: unused

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/InitialDataTci.hpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+struct DgInitialDataTci {
+ private:
+  template <typename Tag>
+  using Inactive = evolution::dg::subcell::Tags::Inactive<Tag>;
+
+ public:
+  using argument_tags = tmpl::list<domain::Tags::Mesh<3>>;
+
+  static bool apply(
+      const Variables<tmpl::list<
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/InitialDataTci.hpp:40]: forward declare Variables

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/InitialDataTci.cpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Evolution/DgSubcell/PerssonTci.hpp"
+#include "Evolution/DgSubcell/TwoMeshRdmpTci.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/InitialDataTci.cpp]: unused

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/InitialDataTci.cpp [hunk 4]
```diff
@@ -0,0 +1,37 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/InitialDataTci.hpp"
+
+#include <cstddef>
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/InitialDataTci.cpp:6]: unused

**Review by kidder — CHANGES_REQUESTED** [review]

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/InitialDataTci.cpp (hunk 4 above)
Comment (nilsdeppe) [reply src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/InitialDataTci.cpp:6]: Mesh takes a size_t as it's argument, though :shrug:

**Comment by nilsdeppe** [comment]
Rebased and squashed in changes since they're quite small. Thanks for the review :)

**Review by kidder — APPROVED** [review]

**Review by fmahebert — APPROVED** [review]

**Pull request merged**